examples/elman/RecurrentLayer.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE FunctionalDependencies #-}
+
+module RecurrentLayer where
+
+import Torch.Tensor
+import Torch.DType
+import Torch.TensorFactories
+import Torch.Functions
+import Torch.TensorOptions
+import Torch.Autograd
+
+import ATen.Cast
+import qualified ATen.Managed.Native as ATen
+
+import System.IO.Unsafe
+import Control.Monad.State.Strict
+import Data.List (foldl', scanl', intersperse)
+
+import Parameters
+import Utils
+
+{- This is the core of the RNN example- an Elman Cell type -}
+
+
+-- Specifying the shape of the recurrent layer
+data RecurrentSpec = RecurrentSpec { in_features :: Int, hidden_features :: Int, nonlinearitySpec :: Tensor -> Tensor }
+
+-- Recurrent layer type holding the weights for the layer
+data Recurrent = Recurrent { weight_ih :: Parameter, bias_ih :: Parameter,
+                             weight_hh :: Parameter, bias_hh :: Parameter
+                        }
+
+
+-- Typeclass that shows that the layer weights can be randomly initialized
+instance Randomizable RecurrentSpec Recurrent where
+  sample RecurrentSpec{..} = do
+      w_ih <- makeIndependent =<< randn' [in_features, hidden_features]
+      b_ih <- makeIndependent =<< randn' [1, in_features]
+      w_hh <- makeIndependent =<< randn' [hidden_features, hidden_features]
+      b_hh <- makeIndependent =<< randn' [1, hidden_features]
```

Comment:
We don't need two biases. It's like that in PyTorch because of cuDNN compatibility, but you don't need to carry over this weird pattern.

Reply:
I'd like to keep the two biases because of pattern matching to other RNN cell types- eg: when I implement a GRU cell, I'll have separate weight/bias pairs for each gate.

Reply:
Oh yeah, each 'gate' in an RNN has a single bias. Changed.